### PR TITLE
Add Relation Metadata

### LINKED ISSUE #2364: Add relation metadata
## Scope & Context

Followup https://github.com/twentyhq/twenty/issues/2191 

## Technical inputs

We should create a new relationMetadata table

```
CREATE TABLE metadata."relationMetadata" (
    id uuid DEFAULT uuid_generate_v4() PRIMARY KEY,
    "parentObjectMetadataId" uuid,
    "childObjectMetadataId" uuid,
    "parentFieldMetadataId" uuid,
    "childFieldMetadataId" uuid,
    "relationType" text,
    "workspaceId" uuid
);
```

We also need a new RELATION type for FieldMetadata.

### PR DESCRIPTION
**Pull request by Weiko**
## Context
Fix https://github.com/twentyhq/twenty/issues/2364

## Technical implementation

To handle relations, we are introducing a new relation-metadata table that should store the type of relation and the different objects involved in the relation.
From the UI, we are creating a new field of type relation, which is also what we do in the backend but instead we create 2 fields, 1 from each side of the relation. As mentioned above the relation, we also create a relation-metadata.

- Introducing a new "RELATION" type
- Renaming a few fields of TenantMigrationColumnActions to add clarity: name -> columnName, type -> columnType and also separating each action in its own type, with the new TenantMigrationColumnRelation having fields necessary to create the FK.
- Adding a new createForeignKey method in the migrationRunner, its responsibility is to only create the foreign key which means the column needs to exist first. (via a TenantMigrationColumnCreate)

Note: The API to create custom relations is not fully working yet since we are focusing on migrating standard objects for now, technically it's possible using `createOneRelation` but I haven't tested everything

### CODE REVIEW

File: server/src/metadata/migration-runner/migration-runner.service.ts [hunk 1]
```diff
@@ -151,10 +153,16 @@ export class MigrationRunnerService {
             columnMigration,
           );
           break;
-        default:
-          throw new Error(
-            `Migration column action ${columnMigration.action} not supported`,
+        case 'relation':
```
Comment (magrinj) [inline server/src/metadata/migration-runner/migration-runner.service.ts:156]: @Weiko For more readability, maybe you can create an enum for your actions so it's easier to read and understand what action are possible

File: server/src/metadata/migration-runner/migration-runner.service.ts [hunk 2]
```diff
@@ -171,22 +179,40 @@ export class MigrationRunnerService {
...
+    tableName: string,
+    migrationColumn: TenantMigrationColumnRelation,
+  ) {
+    await queryRunner.createForeignKey(
+      `${schemaName}.${tableName}`,
+      new TableForeignKey({
+        columnNames: [migrationColumn.columnName],
+        referencedColumnNames: [migrationColumn.referencedTableColumnName],
+        referencedTableName: migrationColumn.referencedTableName,
+        onDelete: 'CASCADE',
```
Comment (magrinj) [inline server/src/metadata/migration-runner/migration-runner.service.ts:214]: @Weiko @charlesBochet Is it something we want to be customizable ?

**Review by magrinj — APPROVED** [review]
Great job ! Few comments :)

File: server/src/metadata/migration-runner/migration-runner.service.ts (hunk 1 above)
Comment (Weiko) [reply server/src/metadata/migration-runner/migration-runner.service.ts]: Good point, done 👍

**Pull request merged**